File: AirYourBnb/src/main/java/AirbnbAnalysis.java
import java.io.Serializable;

import org.apache.spark.SparkConf;

import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.api.java.JavaPairRDD;
import org.apache.spark.api.java.JavaRDD;

import org.apache.spark.sql.SparkSession;
import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.Encoder;
import org.apache.spark.sql.Encoders;
import org.apache.spark.sql.Row;
import scala.Tuple2;

import java.util.Arrays;

public class AirbnbAnalysis{



    //When we submit the job pass our hdfs cluster as a single arg 1 2 or 3
    public static void main(String[] args){

        SparkSession spark = SparkSession
                .builder()
                .appName("Java Spark SQL data sources example")
                .config("spark.some.config.option", "some-value")
                .getOrCreate();

//        SparkConf sparkConf = new SparkConf();//.setMaster("").setAppName("JD Word Counter");
//        JavaSparkContext sparkContext = new JavaSparkContext(sparkConf);


        String fileName = null;
        String anthony_hdfs = "hdfs://dover.cs.colostate.edu:42080/cs455/termproject/airbnb-listings.csv";
        String daniel_hdfs = "hdfs://juneau.cs.colostate.edu:11223/cs455/TERM/airbnb-listings.csv";
        String bruce_hdfs = "hdfs://jackson:2084/airbnb/airbnb-listings.csv";

        if(args.length != 1){
            System.err.println("Supply your HDFS cluster information");
            System.exit(-1);
        }

        switch(Integer.parseInt(args[0])){
            case 1:
                fileName = anthony_hdfs;
                break;
            case 2:
                fileName = daniel_hdfs;
                break;
            case 3:
                fileName = bruce_hdfs;
                break;
        }

        Dataset<Row> csv = spark.read().format("csv").option("header","true").option("delimiter", ";").load(fileName);
        //csv.show();
        csv.write().json("/cs455/termproject/temp-data.json");
//        JavaRDD<String> inputFile = sparkContext.textFile(fileName);
//
//        JavaRDD<String> wordsFromFile = inputFile.flatMap(content -> Arrays.asList(content.split(" ")).iterator());
//
//        JavaPairRDD countData = wordsFromFile.mapToPair(t -> new Tuple2(t, 1)).reduceByKey((x, y) -> (int) x + (int) y);

    }

}

File: AirYourBnb/src/main/scala/AirbnbAnalysis.scala
import java.io.File

import scala.io.Source._

import org.apache.spark.sql.SparkSession

/*

https://stackoverflow.com/questions/40800920/how-do-i-convert-arrayrow-to-dataframe
 */

object AirbnbAnalysis {
  def main(args: Array[String]) {

    import sqlContext.implicits._
    import org.apache.spark.sql.functions._

    val spark = SparkSession
      .builder
      .appName("AirYourBnb")
      .getOrCreate()

    val sparkContext = spark.sparkContext;
    val sqlContext = new org.apache.spark.sql.SQLContext(sparkContext)

    val anthony_hdfs = "hdfs:///cs455/termproject/airbnb-listings.csv"
    val daniel_hdfs = "hdfs:///cs455/TERM/airbnb-listings.csv"
    val bruce_hdfs = "hdfs:///airbnb/out.csv"

    val user = args(0)
    val filename = ""

    user match {
      case 1  => filename = anthony_hdfs
      case 2  => filename = bruce_hdfs
      case 3  => filename = daniel_hdfs
    }

    val df = spark.read.format("org.apache.spark.csv").option("header", true).option("delimiter", ";").csv(filename)

    val table = df.select("Country","Price","Property Type", "Accommodates", "Amenities", "City")

    //This scrapes the table and removes all of the invalid entries that contain null as a value
    val notNull = table.where(df.col("Country").isNotNull).where(df.col("Price").isNotNull).where(df.col("Property Type").isNotNull).where(df.col("Accommodates").isNotNull).where(df.col("Amenities").isNotNull)

    //FIXME
    val lmao = notNull.rdd.flatMap(row =>  row.getString(4).split(",").map(am => ( (row.getString(0),  row.getString(2), am), (row.getString(1).toInt/row.getString(3).toInt, 1) ) ) ).reduceByKey((x:(Int, Int), y:(Int, Int)) => (x._1+y._1, x._2+y._2))
    val hmmst = lmao.map(row => (row._1._1, row._1._2, row._1._3, (row._2._1/row._2._2), row._2._2, row._2._1) )

    val total_listing_types = notNull.rdd.map(row => ( (row.getString(0),  row.getString(2)), ( 1) ) )
        .reduceByKey((x:(Int), y:(Int)) => (x+y))

    spark.stop();
  }
}
